UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -566,8 +572,9 @@ ParsegraphicNode (
 **/
 VOID
 ParseSerialPort (
-  IN VOID   *Fdt,
-  IN INT32  SubNode
+  IN VOID     *Fdt,
+  IN INT32    SubNode,
+  IN BOOLEAN  SerialOnIsa
   )

```

Comment:
I think we could know if it is legacy/ISA serial or PCI serial from SerialPort property 'reg' address. The parameter SerialOnIsa is not required. On the other hand, to share a common ParseSerialPort function for root node and its sub-node, this function need know how to decode 'reg' property which need #address-cells and #size-cells as parameter.

Reply:
I created a new PR for serial port node enhancement. please check if it works for you. https://github.com/tianocore/edk2/pull/6451

Reply:
That is generally a better implementation of this. It doesn't account for cases where `/isa` is in the root, so I'll replace my serial port commit to handle that using #6451. Therefore, I'm marking this as done.